lib/core/src/Cardano/Wallet/Api.hs

```diff
@@ -320,6 +325,29 @@ type DeleteTransaction = "wallets"
     :> Capture "transactionId" ApiTxId
     :> DeleteNoContent
 
+{-------------------------------------------------------------------------------
+                                 Shelley Migrations
+
+  See also: https://input-output-hk.github.io/cardano-wallet/api/#tag/Shelley-Migrations
+-------------------------------------------------------------------------------}
+
+type ShelleyMigrations n =
+         GetShelleyWalletMigrationInfo
+    :<|> MigrateShelleyWallet n
+
+-- | https://input-output-hk.github.io/cardano-wallet/api/#operation/migrateShelleyWallet
+type MigrateShelleyWallet n = "wallets"
+    :> Capture "walletId" (ApiT WalletId)
+    :> "migrations"
+    :> ReqBody '[JSON] (ApiWalletMigrationPostDataT n "raw")
```

Comment:
It is passed as the `Passphrase (purpose :: Symbol)` type parameter. "raw" now is used widely to mean a normal passphrase that is not "lenient". We could surely come up with a better name.

Reply:
because of "legacy" (mainly because of decisions taken by daedalus previous versions) we use for byron wallets "lenient" (ie., passwords lengths can be from 0 to 255 ). For shelley we do proper passwords (ie. accept those having (10,255) characters), and they are named "raw". In swagger this is represented by lenientPassword and walletPassword, respectively.

Reply:
@rvl, @paweljakubas: Thanks for the explanations. :+1: 

Is there anywhere in the source code where we document these symbols?

**Just thinking out loud here**: instead of using string symbols, could we (hypothetically) replace these with types?

For example:

```hs
-- | Indicates the purpose of a passphrase.
data PassphrasePurpose
    = Lenient
        -- ^ Explanation of what it means to be a "lenient" passphrase.
    | Raw
        -- ^ Explanation of what it means to be a "raw" passphrase.
    | Encryption
        -- ^ Explanation of what it means to be an "encryption" passphrase.
```
Then, functions that take and return passphrases can have types like:
```hs
preparePassphrase
    :: PassphraseScheme
    -> Passphrase `Raw
    -> Passphrase `Encryption 
preparePassphrase = \case    
```

If someone is using source code introspection tools (or even just Haddock), then navigating to the definition of `Raw` will lead to an explanation of what it means to be a "raw" passphrase. (That is, assuming we've written such an explanation.)

Thoughts?

(Obviously, this is outside the scope of this PR.)

Reply:
discusses amidst group members - we will tackle it in some next PR

Reply:
> discusses amidst group members - we will tackle it in some next PR

@paweljakubas I created an issue here: https://jira.iohk.io/browse/ADP-328